cpp/arcticdb/version/version_core.cpp

```diff
@@ -1456,6 +1456,14 @@ VersionedItem sort_merge_impl(
         "Finalizing staged data is not allowed with empty staging area"
     );
 
+    user_input::check<ErrorCode::E_INVALID_USER_ARGUMENT>(
+        write_options.dynamic_schema || pipeline_context->staged_descriptor_->field_count() < write_options.column_group_size,
+        "Sorting and finalizing staged data is not implemented in the case when column slicing would appear. The "
```

Comment:
not yet implemented?

we do have a plan to solve this in future

Reply:
In the long term, yes. It would require a large refactor though.